frontend/src/bundles/common/components/calendar/calendar.tsx

```diff
@@ -0,0 +1,186 @@
+    name,
+    control
+}: Properties<T>): JSX.Element => {
+    const { control: innerControl } = useAppForm<CalendarDate>({
+        defaultValues: DEFAULT_DATE_PAYLOAD,
+    });
+
+    const { field } = useFormController({ name, control });
+
+    const { field: presentField } = useFormController({ name: 'present', control: innerControl });
+
+    const reference = useRef<HTMLDivElement>(null);
+
+    const [text, setText] = useState('');
+
+    const handleTextChange = useCallback((error: ChangeEvent<HTMLInputElement>): void => {
+        setText(error.target.value);
+
+        const year = error.target.value.match(yearRegex);
+        if (year) {
+            setYear(Number(year[0]));
+        }
+
+        const monthMatch = error.target.value.match(monthRegex);
+
+        if (monthMatch) {
+            for(const month of CalendarMonths) {
+                if (
+                    monthMatch[0]
+                        .toLowerCase()
+                        .includes(month.name.toLowerCase()) ||
+                    month.name
+                        .toLowerCase()
+                        .startsWith(monthMatch[0].toLowerCase())
+                ) {
+                    setMonth(month);
+                    setSelected(month.num);
+                }
+            }
+        } else {
+            setMonth(null);
+            setSelected(0);
+        }
+    }, []);
+
+    const [focused, setFocused] = useState(false);
+
+    const handleInputFocus = useCallback((focused: boolean) => 
+        clsx(styles.calendar__date_input, styles.focused, focused && styles.calendar__date_input)
+    , []);
+
+    const setCurrentlyFocused = useCallback((): void => setFocused(true), []);
+    const setUnfocused = useCallback((): void => setFocused(false), []);
+
+    const [selected, setSelected] = useState<number | null>(null);
+
+    const [year, setYear] = useState(new Date().getFullYear());
+    const [month, setMonth] = useState<CalendarMonth | null>(null);
+
+    const increaseYear = useCallback((): void => setYear(year + 1), [year]);
+    const decreaseYear = useCallback((): void => setYear(year - 1), [year]);
+
+    const selectMonth = useCallback((value: Date): void => {
+        const inputMonth = CalendarMonths.find((month) => month.num === value.getMonth());
+        if(inputMonth) {
+            setMonth(inputMonth);
+            setSelected(inputMonth.num);
+        }
+    }, []);
+
+    const handleMonthSelect = useCallback(({ date }: { date: Date }) => 
+        clsx(styles.date_picker__option, date.getMonth() === month?.num && styles.date_picker__selected)
+    , [month]);
+
+    const handleYearSelected = useCallback((selected: number | null) => 
+        clsx(styles.date_picker__option, styles.date_picker__option_year, selected === null && styles.date_picker__selected)
+    , []);
+    
+    const selectYear = useCallback((): void => {
+        setMonth(null);
+        setSelected(null);
+    }, []);
+
+    useEffect(() => {
```

Comment:
I would move this logic to a separate hook

Reply:
done